packages/data-stores/src/products-list/actions.ts

```diff
@@ -0,0 +1,53 @@
+import { ensureNumericCost } from './assembler';
+import {
+	PRODUCTS_LIST_RECEIVE,
+	PRODUCTS_LIST_REQUEST,
+	PRODUCTS_LIST_REQUEST_FAILURE,
+} from './constants';
+import type { ProductsList, ProductsListFailure, ProductsListResponse } from './types';
+
+export function createActions() {
+	const receiveProductsList = (
+		productsList: ProductsList,
+		type: string | null = null
+	): ProductsListResponse => {
+		// Since the request succeeded, productsList should be guaranteed non-null;
+		// thus, we don't have any safety checks before this line.
+
+		// Create a completely new products list to avoid mutating the original
+		const sanitizedProductsList = Object.fromEntries(
+			Object.entries( productsList ).map( ( [ slug, product ] ) => [
+				slug,
+				ensureNumericCost( product ),
+			] )
+		);
+
+		return {
+			type: PRODUCTS_LIST_RECEIVE,
+			productsList: sanitizedProductsList,
+			productsListType: type,
+		};
+	};
+
+	const requestProductsList = () => ( {
+		type: PRODUCTS_LIST_REQUEST,
+	} );
+
+	const receiveProductsListFailure = ( error: ProductsListFailure ) => ( {
+		type: PRODUCTS_LIST_REQUEST_FAILURE,
+		error,
+	} );
+
+	return { receiveProductsList, receiveProductsListFailure, requestProductsList };
+}
```

Comment:
`receiveProductsList` is a function because it needs to do some additional processing to build the return object (that's functionality I pulled over from the `client/state` code.

But `requestProductsList` definitely doesn't!

Reply:
Sorry, I meant `createActions`. These functions can be inlined.

Reply:
That's how it was done in the `site` data-store which is what I've been using as an example.

But I see how it's done in the `onboarding` data-store and that's much tidier. I can tweak it to use that approach instead.